src/components/Notification/Notification.tsx

```diff
@@ -0,0 +1,201 @@
+export function NotificationContent({ ...props }) {
+  const { children, className } = props;
+  return (
+    <div className={bem("notification-content", [], "", [className])}>
+      {children}
+    </div>
+  );
+}
+
+export default function Notification(
+  props: React.PropsWithChildren<NotificationProps>
+) {
+  const {
+    blockName,
+    centered,
+    children,
+    className,
+    dismissible = false,
+    noMargin = false,
+    icon,
+    id,
+    modifiers = [],
+    notificationType = "standard",
+  } = props;
+
+  const [isOpen, setIsOpen] = useState(true);
+
+  if (!isOpen) {
+    return null;
+  }
+
+  const handleClose = () => setIsOpen(false);
+
+  const baseClass = "notification";
+
+  const notificationModifiers = modifiers ? modifiers : [];
+
+  notificationModifiers.push(notificationType);
+
+  if (centered) notificationModifiers.push("centered");
+
+  if (noMargin) notificationModifiers.push("no-margin");
+
+  const iconElement = () => {
+    if (icon) return icon;
+    switch (notificationType) {
+      // just using placeholders until the necessary icons are added
+      case "announcement":
+        return (
+          <Icon
+            decorative={false}
+            className="notification-icon"
+            name={IconNames.speaker_notes}
+            color={IconColors.section_research_secondary}
+            modifiers={["large"]}
+          />
+        );
+      case "warning":
+        return (
+          <Icon
+            decorative={false}
+            className="notification-icon"
+            name={IconNames.error_filled}
+            color={IconColors.brand_primary}
+            modifiers={["large"]}
+          />
+        );
+      default:
+        return null;
+    }
+  };
+
+  let childHeading,
+    childContent,
+    headingCount = 0,
+    contentCount = 0;
+
+  React.Children.map(children, (child: React.ReactElement) => {
+    if (
+      child.type === NotificationHeading ||
+      child.props.mdxType === "NotificationHeading"
+    ) {
+      childHeading = child;
+      headingCount++;
+    }
+
+    if (
+      child.type === NotificationContent ||
+      child.props.mdxType === "NotificationContent"
+    ) {
+      childContent = child;
+      contentCount++;
+    }
+  });
+
+  if (headingCount > 1) {
+    console.error(
+      `Only one NotificationHeading child component may be passed to Notification.`
+    );
+    childHeading =
+      "Error: Only one NotificationHeading child component may be passed to Notification.";
+  }
+
+  if (contentCount > 1) {
+    console.error(
+      `Only one NotificationContent child component may be passed to Notification.`
+    );
+    childContent =
+      "Error: Only one NotificationContent child component may be passed to Notification.";
+  }
+
+  if (
+    headingCount &&
+    (icon ||
+      ((notificationType === NotificationTypes.Announcement ||
+        notificationType === NotificationTypes.Warning) &&
+        !centered))
+  )
```

Comment:
Nit: for multiline `if` statements, add the curly braces around line 164.

Reply:
Done.